psi4/driver/json_wrapper.py

```diff
@@ -169,70 +345,47 @@ def run_json(json_data):
             return json_data
 
     # Check driver call
-    if json_data["driver"] not in list(methods_dict):
+    if json_data["driver"] not in list(methods_dict_):
         json_data["error"] = "Driver parameters '%s' not recognized" % str(json_data["driver"])
         return json_data
 
-    # Set scratch
-    if "scratch_location" in list(json_data):
-        psi4_io = core.IOManager.shared_object()
-        psi4_io.set_default_path(json_data["scratch_location"])
+    # Set options
+    if "options" in json_data.keys() and (json_data["options"] is not None):
+        for k, v in json_data["options"].items():
+            core.set_global_option(k, v)
 
-    # Set memory
-    if "memory" in list(json_data):
-        psi4.set_memory(json_data["memory"])
+    # Rework args
+    args = json_data["method"]
+    if not isinstance(args, (list, tuple)):
+        args = [args]
 
-    # Do we return the output?
-    return_output = json_data.pop("return_output", False)
-    if return_output:
-        outfile = str(uuid.uuid4()) + ".json_out"
-        core.set_output_file(outfile, False)
-        json_data["raw_output"] = "Not yet run."
+    # Deep copy kwargs
+    if "kwargs" in list(json_data):
+        kwargs = copy.deepcopy(json_data["kwargs"])
+    else:
+        kwargs = {}
 
-    try:
-        # Set options
-        if "options" in json_data.keys() and (json_data["options"] is not None):
-            for k, v in json_data["options"].items():
-                core.set_global_option(k, v)
+    kwargs["molecule"] = molutil.geometry(json_data["molecule"])
 
-        # Rework args
-        args = json_data["method"]
-        if not isinstance(args, (list, tuple)):
-            args = [args]
+    # Full driver call
+    kwargs["return_wfn"] = True
 
-        # Deep copy kwargs
-        if "kwargs" in list(json_data):
-            kwargs = copy.deepcopy(json_data["kwargs"])
-        else:
-            kwargs = {}
+    if json_data["driver"] not in methods_dict_:
+        raise KeyError("Driver type '%s' not recognized." % str(json_data["driver"]))
 
-        kwargs["molecule"] = molutil.geometry(json_data["molecule"])
+    val, wfn = methods_dict_[json_data["driver"]](*args, **kwargs)
```

Comment:
I'll need to define an extension for add'l kwargs, yes?

Reply:
Yeah, add'l kwargs and the ability to record whole output record goes down dramatically. But for the getting-things-running aspect of the separated driver, I'll need it. record aspect comes later.